R/get_metadata.R

```diff
@@ -1,26 +1,15 @@
 #' Get Water System
-#' @description Enable user to discover water system ID when supplied a zipcode or county of interest
-#' @param zipcode zipcode
+#' @description Enable user to discover water system ID when supplied a county of interest
 #' @param county county
 #' @examples
-#' get_water_system(zipcode = "90201")
 #' get_water_system(county = "D")
 #' @export
-get_water_system <- function(...) {
+get_water_system <- function(county) {
 
-  arg <- list(...)
-  if (length(arg) == 0) stop("supply either a zipcode or county", call. = FALSE)
-  if (is.null(names(arg))) stop("supply named argument", call. = FALSE)
-  if (length(arg) > 1) stop("supply only zipcode or county", call. = FALSE)
-
-  if (names(arg) == "zipcode") {
-    water_system <- water_systems[water_systems$zipcode == arg$zipcode, ]$water_system_name
-  } else if (names(arg) == "county") {
-    water_system <- water_systems[water_systems$county == arg$county, ]$water_system_name
-  }
+  water_system <- water_systems[water_systems$county == county, ]$water_system_name
```

Comment:
It seems this function errors if the user doesn't know how the county is exactly spelled in the `water_systems` object. Perhaps if the user inputs a water system that's not in the valid counties (e.g., misspelling), we throw a more informative error and prompt the user to select a valid county: `paste("The input county is invalid, perhaps you mean one of these:", paste(unique(water_systems$county), collapse=", ")`".

Reply:
We could have a county look up function to help people identify proper spelling

Reply:
Yes! We can easily achieve this with an approximate grep, `agrep()`.

Here's an example where the user inputs "VETURA" instead of "VENTURA":

```r
user_input_county <- "VETURA"
counties <- unique(water_systems$county)

if(! user_input_county %in% counties) {
  approximate_match <- counties[ agrep(user_input_county, counties) ]
  stop(paste0("Couldn't find ", user_input_county, " but did you mean ", approximate_match, "?"), call. = FALSE)
} 
```